src/test/java/com/danit/controllers/employee/TrainingTypeControllerTest.java

```diff
@@ -0,0 +1,164 @@
+    HttpHeaders header = testUtils.getHeader(template, UserRolesEnum.USER);
+    this.mockMvc.perform(post(url).headers(header)
+        .contentType("application/json")
+        .content("{\n"
+            + "    \"name\": \"Test trainingType\",\n"
+            + "    \"description\": \"Test trainingType\"\n"
+            + "  }"))
+        .andExpect(status().isOk());
+
+    mockMvc.perform(get(url).headers(header))
+        .andExpect(content().contentType(MediaType.APPLICATION_JSON_UTF8))
+        .andExpect(jsonPath("$", hasSize(currentQuant + 1)));
+  }
+
+  @Test
+  public void updateTrainingTypeTest() throws Exception {
+    HttpHeaders header = testUtils.getHeader(template, UserRolesEnum.USER);
+
+    String responseJson = this.mockMvc.perform(post(url).headers(header)
+        .contentType("application/json")
+        .content("{\n"
+            + "    \"name\": \"Test trainingType\",\n"
+            + "    \"description\": \"Test trainingType\"\n"
+            + "  }"))
+        .andExpect(status().isOk())
+        .andReturn().getResponse().getContentAsString();
+
+    ObjectMapper mapper = new ObjectMapper();
+    TrainingType actualObj = mapper.readValue(responseJson, new TypeReference<TrainingType>() {
+    });
+    long createdId = actualObj.getId();
+    System.out.println(actualObj);
+
+    responseJson = mockMvc.perform(put(url+"/" + createdId).headers(header)
+        .contentType("application/json")
+        .content("{\n"
+            + "    \"id\": " + createdId + ", \n"
+            + "    \"name\": \"TestTrainingType2\"\n"
+            + "  }"))
+        .andExpect(status().isOk())
+        .andReturn().getResponse().getContentAsString();
+
+    actualObj = mapper.readValue(responseJson, new TypeReference<TrainingType>() {
+    });
+    System.out.println(actualObj);
+    assertEquals("TestTrainingType2", actualObj.getName());
+
+  }
+
+  @Test
+  public void createTrainingTypeTest() throws Exception {
+    int currentQuant = trainingTypeService.getTrainingTypeQuant();
+    HttpHeaders header = testUtils.getHeader(template, UserRolesEnum.USER);
+    this.mockMvc.perform(post(url).headers(header)
+        .contentType("application/json")
+        .content("{\n"
+            + "    \"name\": \"Test trainingType\",\n"
+            + "    \"description\": \"Test trainingType\"\n"
+            + "  }"))
+        .andExpect(status().isOk());
+
+    mockMvc.perform(get(url).headers(header))
+        .andExpect(content().contentType(MediaType.APPLICATION_JSON_UTF8))
+        .andExpect(jsonPath("$", hasSize(currentQuant + 1)));
+  }
+
+  @Test
+  public void expectIsNotFoundWhenNoDataFoundService() throws Exception {
+    HttpHeaders header = testUtils.getHeader(template, UserRolesEnum.USER);
+
+    mockMvc.perform(get(url+"/0").headers(header))
+        .andExpect(status().isNotFound());
+  }
+
+  @Test
+  public void expect500WhenDeleteNonexistentService() throws Exception {
```

Comment:
expect500WhenDeleteNonexistentTrainingType
or could be 
expect404WhenDeleteNonexistentTrainingType
if you agree on returning 404 for deleting non-existing objects

Reply:
delete returns 500 if object doesn't exists